src/github/github.service.ts

```diff
@@ -0,0 +1,167 @@
+import {
+  BadRequestException,
+  HttpException,
+  HttpStatus,
+  Injectable,
+  Logger,
+  NotFoundException,
+} from '@nestjs/common';
+import { Octokit } from '@octokit/rest';
+import { EntityManager, In, Repository } from 'typeorm';
+import { InjectEntityManager, InjectRepository } from '@nestjs/typeorm';
+import { User } from './entities/user.entity';
+import { ApiMessages, TokenUsageTags } from 'src/common/constants';
+import { TRequest } from 'src/common/types/user.types';
+import { Env } from 'src/env';
+import * as fs from 'fs';
+import * as jwt from 'jsonwebtoken';
+import * as path from 'path';
+import axios from 'axios';
+import { Repo } from 'src/webhook/entities/repository.entity';
+import { UpdatePracticesDto } from './dto/update.practices.dto';
+import { OpenAiService } from 'src/open-ai/open-ai.service';
+import { calculateOffset } from 'src/utils/app.utils';
+import { PullRequest } from 'src/webhook/entities/pull-request.entity';
+import { UserInstallation } from './entities/user-installation.entity';
+import { GithubInstallationService } from './github.installation.service';
+
+@Injectable()
+export class GithubService {
+  private readonly logger = new Logger(GithubService.name);
+  constructor(
+    @InjectRepository(User)
+    private userRepository: Repository<User>,
+    @InjectRepository(UserInstallation)
+    private userInstallationRepository: Repository<UserInstallation>,
+    @InjectEntityManager()
+    private entityManager: EntityManager,
+    private readonly openAiService: OpenAiService,
+    private readonly githubInstallationService: GithubInstallationService,
+  ) {}
+
+  async getUserDetails(accessToken: string) {
+    try {
+      const octokit = new Octokit({ auth: accessToken });
+
+      const { data: user } = await octokit.rest.users.getAuthenticated();
+      if (!user) {
+        throw new HttpException('Invalid token', HttpStatus.BAD_REQUEST);
+      }
+
+      const { data: emails } =
+        await octokit.rest.users.listEmailsForAuthenticatedUser();
+      const primaryEmail = emails.find((email) => email.primary)?.email;
+
+      return {
+        ...user,
+        email: primaryEmail || null,
+      };
+    } catch (error) {
+      if (error.status) throw new HttpException(error.message, error.status);
+      else throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+    }
```

Comment:
PullProofAI

Consider handling specific error types instead of catching all errors in a generic catch block. This will provide better error messages and help in debugging.



<!-- This is an auto-generated comment by PullProofAI -->